h5t.go

```diff
@@ -466,6 +466,9 @@ func NewDataTypeFromType(t reflect.Type) (*Datatype, error) {
 		}
 		dt = &cdt.Datatype
 
+	case reflect.Ptr:
+		return NewDataTypeFromType(t.Elem())
```

Comment:
if `t` is `*int`, this will return the HDF5 DataType for `int`.
shouldn't we instead return the HDF5 DataType that describes a pointer to `int` ?

Reply:
This type specification is in line with the current implementation of Append:
https://github.com/gonum/hdf5/blob/master/h5pt.go#L193-L195
```
	case reflect.Ptr:
		ptrVal := rp.Elem()
		cData = unsafe.Pointer(&ptrVal)
```
If you propose to use `H5T_STD_REF_OBJ` as a type of `*int` and then add the actual `int` to a separate file location the `Append` function (possibly among many others) would need to be refactored as well.
The change as is now, prevents a panic in `default` and works as intended.

Reply:
I fixed the test cases. I think they are testing now the stuff that should be tested.